homeassistant/components/knx/cover.py

```diff
@@ -73,99 +87,95 @@ def supported_features(self):
 
-    async def async_set_cover_position(self, **kwargs):
+    async def async_set_cover_position(self, **kwargs: Any) -> None:
         """Move the cover to a specific position."""
         knx_position = 100 - kwargs[ATTR_POSITION]
         await self._device.set_position(knx_position)
 
-    async def async_stop_cover(self, **kwargs):
+    async def async_stop_cover(self, **kwargs: Any) -> None:
         """Stop the cover."""
         await self._device.stop()
         self.stop_auto_updater()
 
     @property
-    def current_cover_tilt_position(self):
+    def current_cover_tilt_position(self) -> Optional[int]:
         """Return current tilt position of cover."""
         if not self._device.supports_angle:
             return None
-        try:
-            return 100 - self._device.current_angle()
-        except TypeError:
-            return None
+        ang = self._device.current_angle()
+        return 100 - ang if ang is not None else None
 
-    async def async_set_cover_tilt_position(self, **kwargs):
+    async def async_set_cover_tilt_position(self, **kwargs: Any) -> None:
         """Move the cover tilt to a specific position."""
         knx_tilt_position = 100 - kwargs[ATTR_TILT_POSITION]
         await self._device.set_angle(knx_tilt_position)
 
-    async def async_open_cover_tilt(self, **kwargs):
+    async def async_open_cover_tilt(self, **kwargs: Any) -> None:
         """Open the cover tilt."""
         await self._device.set_short_up()
 
-    async def async_close_cover_tilt(self, **kwargs):
+    async def async_close_cover_tilt(self, **kwargs: Any) -> None:
         """Close the cover tilt."""
         await self._device.set_short_down()
 
-    async def async_stop_cover_tilt(self, **kwargs):
+    async def async_stop_cover_tilt(self, **kwargs: Any) -> None:
         """Stop the cover tilt."""
         await self._device.stop()
         self.stop_auto_updater()
 
-    def start_auto_updater(self):
+    def start_auto_updater(self) -> None:
         """Start the autoupdater to update Home Assistant while cover is moving."""
-        if self._unsubscribe_auto_updater is None:
+        if self._unsubscribe_auto_updater is None and self.hass is not None:
             self._unsubscribe_auto_updater = async_track_utc_time_change(
                 self.hass, self.auto_updater_hook
             )
 
-    def stop_auto_updater(self):
+    def stop_auto_updater(self) -> None:
         """Stop the autoupdater."""
         if self._unsubscribe_auto_updater is not None:
             self._unsubscribe_auto_updater()
             self._unsubscribe_auto_updater = None
 
     @callback
-    def auto_updater_hook(self, now):
+    def auto_updater_hook(self, now: datetime) -> None:
         """Call for the autoupdater."""
         self.async_write_ha_state()
         if self._device.position_reached():
+            assert self.hass is not None
+            self.hass.async_create_task(self._device.auto_stop_if_necessary())
```

Comment:
previously `hass.add_job()` was used here. As there is a Coroutine called, I think `async_create_task()` is more appropriate.
https://github.com/XKNX/xknx/blob/6a9be156132bac5233a0a5000e8ea953369d642d/xknx/devices/cover.py#L297
Also the task doesn't need to be created every second if `self._device.position_reached()` was false - this check is part of auto_stop_if_necessary() too.